### PR TITLE
Stop using `NSFullSizeContentViewWindowMask` to get borderless windows, because that disables the swap interval.

### PR DESCRIPTION
**Pull request by pcwalton**
This hack is very simple but a little evil. We get the superview of the
content view, which is the `NSThemeFrame`, and install an OpenGL context
into it.  `NSThemeFrame` is a private class, but we never speak its
name, we get to it with public APIs, and we only call public `NSView`
APIs on it, so this seems OK in terms of being supported by Apple going
forward.

Additionally, we remove the standard window buttons so that they
disappear and the user can't click them. This can also be done using
public APIs.

The reason for using this hack as opposed to
`NSFullSizeContentViewWindowMask` is that the latter forces the window
to be Core Animation-backed, which results in us rendering to an off
screen surface. Not only does this inject another compositor into the
system, but it also results in us rendering to an off-screen surface,
disabling the swap interval.

This depends on a `cocoa-rs` upgrade to add a binding to the `-[NSView
superview]` method.

Note that the top edge of the window is not rounded, although the shadow
is. Applications that wish to use this mode will need to round the top
edge themselves if they wish. (This is how Cocoa internally works:
rounding is done by the app, not the window manager.)

This should fix servo/servo#9431.

### CODE REVIEW

File: src/api/cocoa/mod.rs [hunk 1]
```diff
@@ -416,8 +416,7 @@ impl Window {
                 NSClosableWindowMask as NSUInteger |
                 NSMiniaturizableWindowMask as NSUInteger |
                 NSResizableWindowMask as NSUInteger |
-                NSTitledWindowMask as NSUInteger |
-                NSFullSizeContentViewWindowMask as NSUInteger
+                NSTitledWindowMask as NSUInteger,
```
Comment (tomaka) [inline src/api/cocoa/mod.rs]: Travis complains that this coma shouldn't be there.

**Comment by tomaka** [comment]
I'm not qualified to review this! If someone who knows cocoa wants to have a look at it, that would be nice.
Otherwise I'm just going to be merge in a few days.

**Comment by paulrouget** [comment]
This has been reviewed by someone who knows cocoa here: https://github.com/servo/glutin/pull/68#issuecomment-188277073

**Comment by ZeGentzy** [comment]
No longer relevant.

**Pull request closed without merging**